streams/src/main/java/org/apache/kafka/streams/state/ReadOnlyKeyValueStore.java

```diff
@@ -58,6 +58,14 @@
      * @throws InvalidStateStoreException if the store is not initialized
      */
     KeyValueIterator<K, V> all();
+    
+    /**
+     * Return an iterator over all keys that share the same prefix byte[] 
+     * as the provided key. 
+     * 
+     */
+    
+    KeyValueIterator<K, V> prefixScan(K prefix);
```

Comment:
@guozhangwang this is controversal and needs discussion. We use the same type for prefix and key. Currently I rely on the Serde to only render a prefix. that works fantastic with protobuf,avro and String might be the worst nightmare for JSON people